doc/source/guide/in_depth/functional_guide.rst

```diff
@@ -0,0 +1,257 @@
+.. _functional_in_depth:
+
+#####################
+Functional
+#####################
+
+A *functional* is an operator ``f`` that maps from some vector space ``X`` to the field of scalars ``F`` associated with the vector space:
+
+.. math::
+   
+   f : X \to F.
+
+In the ODL solver package, functionals are implemented in the `Functional` class, a subclass to `Operator`.
+
+From a mathematical presepective, the above is a valide definition of a functional.
+However, since the purpose of these functionals are primarely to be used for solving optimization problems, the following assumptions are made:
+
+ * the vector space ``X`` is a Hilbert space.
+ * the field of scalars ``F`` are the real numbers.
```

Comment:
I think the concept of convex conjugate breaks down, at least in the current form. You have
```
S^*(f^*) = sup_f { <f^*, f> - S(f) }
```
Here the `S(f)` part is fine since it maps to `R`, but the `<f^*, f>` thing is complex in general, so the `sup` is not well -defined. I've seen definitions of the conjugate using the real part of the `{ }` expression, but I have no idea what part of the theory carries through. The literature is not very extensive on that part.

That said, real-valued functionals on complex spaces can always be seen as functionals on real product spaces. **However**, this gives a different interpretation from the "use real part" above. In the norm example, speaking in terms of product spaces, you take the `sup` only in the first component if you use the real part only, while in the other case, you optimize over both components.